src/new/sendTransaction.js

```diff
@@ -172,15 +276,72 @@ class SendTransaction extends EventEmitter {
     }).catch(() => {
       this.updateOutputSelected(false);
       this.emit('send-error', this.unexpectedPushTxError);
-    });;
+    });
   }
 
   /**
-   * Start object (submit job)
+   * Run sendTransaction from mining, i.e. expects this.transaction to be prepared and signed
+   * then it will mine and push tx
+   *
+   * 'until' parameter can be 'mine-tx', in order to only mine the transaction without propagating
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  runFromMining(until = null) {
+    try {
+      this.mineTx();
+      if (until === 'mine-tx') {
+        return;
+      }
+
+      this.on('mine-tx-ended', (data) => {
+        this.handlePushTx();
+      });
+    } catch (err) {
+      if (err instanceof WalletError) {
+        this.emit('send-error', err);
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  /**
+   * Method created for compatibility reasons
+   * some people might be using the old facade and this start method just calls runFromMining
+   *
+   * @memberof SendTransaction
+   * @inner
    */
   start() {
-    this.updateOutputSelected(true);
-    this.mineTransaction.start();
+    this.runFromMining();
+  }
+
+  /**
+   * Run sendTransaction from preparing, i.e. prepare, sign, mine and push the tx
+   *
+   * 'until' parameter can be 'prepare-tx' (it will stop before signing the tx),
+   * or 'mine-tx' (it will stop before send tx proposal, i.e. propagating the tx)
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  run(until = null) {
```

Comment:
Shouldn't it return a promise that will be resolved or rejected accordingly?

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/221/commits/43cdd56eebf3e443132890aaddd495d24aaf24f5

Reply:
That's great! Thanks!